webapp/src/components/Controls/DatasetSplitToggler.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { fireEvent, screen, within } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import { DatasetSplitName } from "types/api";
+import DatasetSplitToggler from "./DatasetSplitToggler";
+import { setupServer } from "msw/node";
+import {
+  getDatasetInfo,
+  getDatasetInfoWithoutEvaluationSplit,
+  getDatasetInfoWithoutTrainSplit,
+} from "mocks/api/mockDatasetInfoAPI";
+
+const onChangeMock = (value: DatasetSplitName) => jest.fn();
+const renderDataSplitToggler = (value: DatasetSplitName) =>
+  renderWithRouterAndRedux(
+    <DatasetSplitToggler value={value} onChange={onChangeMock} />,
+    { route: "/local", path: "/:jobId" }
+  );
+
+describe("DatasetInfo API", () => {
+  const handlers = [getDatasetInfo];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  test("shoud have 2 toggle buttons displayed and enabled with icons", () => {
+    renderDataSplitToggler("eval");
+    expect(screen.queryAllByRole("button")).toHaveLength(2);
+    expect(screen.queryByText("Evaluation Set")).toBeInTheDocument();
+    expect(screen.queryByTestId("SpeedIcon")).toBeInTheDocument();
+    expect(screen.queryByText("Training Set")).toBeInTheDocument();
+    expect(screen.queryByTestId("ModelTrainingIcon")).toBeInTheDocument();
+    expect(
+      screen.getAllByRole("button", { name: /Evaluation/i })[0]
+    ).toHaveAttribute("aria-pressed", "true");
+  });
+
+  test("should the toggle button Evaluation set displayed with SpeedIcon", () => {
+    renderDataSplitToggler("eval");
+    expect(within(screen.getByText("Evaluation Set")).findByRole("SpeedIcon"));
```

Comment:
`findBy` returns a promise, which is not `await`ed here, so this test would pass even if we changed it to
```suggestion
    expect(within(screen.getByText("Evaluation Set")).findByRole("potato"));
```
However, using `getBy`, this would expectedly fail
```suggestion
    expect(within(screen.getByText("Evaluation Set")).getByRole("SpeedIcon"));
```
That is because the `SpeedIcon` is not a role. I think you can use `getByTestId` instead:
```suggestion
    expect(within(screen.getByText("Evaluation Set")).getByTestId("SpeedIcon"));
```
Same thing for next test.

General comment: When writing a test, we have to make sure it actually tests something. Here, as I showed for example, I would switch `SpeedIcon` with `potato` to make sure the test fails. In more complex situations, I would consciously change the component we are testing to make it fail (for example here changing the icon in `DataSplitToggler`). If the test still passes, it's that it doesn't do its job.